File: src/main/java/org/steamproject/model/Player.java
package org.steamproject.model;

/**
 * Représente un joueur sur la plateforme.
 * Contient les informations protégées par le RGPD.
 */
public class Player {
    private String id;
    private String username;            // Pseudo unique (RGPD)
    private String email;
    private String registrationDate;
    // === Champs RGPD ===
    private String firstName;           // Prénom (RGPD)
    private String lastName;            // Nom (RGPD)
    private String dateOfBirth;         // Date de naissance ISO 8601 (RGPD)
    private Boolean gdprConsent;        // Consentement RGPD accepté
    private String gdprConsentDate;     // Date du consentement RGPD
    // === Statistiques ===
    private Integer totalPlaytime;
    private String lastEvaluationDate;
    private Integer evaluationsCount;
    private java.util.List<GameOwnership> library;

    public Player() {}

    public java.util.List<GameOwnership> getLibrary() { return library; }
    public void setLibrary(java.util.List<GameOwnership> library) { this.library = library; }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getRegistrationDate() { return registrationDate; }
    public void setRegistrationDate(String registrationDate) { this.registrationDate = registrationDate; }

    public Integer getTotalPlaytime() { return totalPlaytime; }
    public void setTotalPlaytime(Integer totalPlaytime) { this.totalPlaytime = totalPlaytime; }

    public String getLastEvaluationDate() { return lastEvaluationDate; }
    public void setLastEvaluationDate(String lastEvaluationDate) { this.lastEvaluationDate = lastEvaluationDate; }

    public Integer getEvaluationsCount() { return evaluationsCount; }
    public void setEvaluationsCount(Integer evaluationsCount) { this.evaluationsCount = evaluationsCount; }

    
    public String getFirstName() { return firstName; }
    public void setFirstName(String firstName) { this.firstName = firstName; }

    public String getLastName() { return lastName; }
    public void setLastName(String lastName) { this.lastName = lastName; }

    public String getDateOfBirth() { return dateOfBirth; }
    public void setDateOfBirth(String dateOfBirth) { this.dateOfBirth = dateOfBirth; }

    public Boolean getGdprConsent() { return gdprConsent; }
    public void setGdprConsent(Boolean gdprConsent) { this.gdprConsent = gdprConsent; }

    public String getGdprConsentDate() { return gdprConsentDate; }
    public void setGdprConsentDate(String gdprConsentDate) { this.gdprConsentDate = gdprConsentDate; }

    @Override
    public String toString() {
        return "Player{" +
                "id='" + id + '\'' +
                ", username='" + username + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", email='" + email + '\'' +
                ", dateOfBirth='" + dateOfBirth + '\'' +
                ", totalPlaytime=" + totalPlaytime +
                '}';
    }
}

File: src/main/kotlin/model/Player.kt
package org.example.model

/**
 * Représente un joueur inscrit sur une plateforme de distribution.
 * Contient les informations protégées par le RGPD.
 * 
 * Ceci reflète la réalité du marché où un joueur a un compte sur chaque plateforme
 * et possède des jeux distincts sur chacune.
 */
data class Player(
    val id: String,
    val username: String,               // Pseudo unique (RGPD)
    val email: String,
    val registrationDate: String,
    
    /**
     * Identifiant de la plateforme de distribution sur laquelle le joueur est inscrit.
     */
    val distributionPlatformId: String? = null,
    
    // === Champs RGPD ===
    val firstName: String? = null,      // Prénom (RGPD)
    val lastName: String? = null,       // Nom (RGPD)
    val dateOfBirth: String? = null,    // Date de naissance ISO 8601 (RGPD)
    val gdprConsent: Boolean = false,   // Consentement RGPD accepté
    val gdprConsentDate: String? = null,// Date du consentement RGPD
    // === Bibliothèque et statistiques ===
    val library: List<GameOwnership> = emptyList(),
    val totalPlaytime: Int? = null,     // en heures, rempli via Kafka
    val lastEvaluationDate: String? = null,
    val evaluationsCount: Int? = null
) {
    /**
     * Retourne la plateforme de distribution associée à ce joueur.
     * Par défaut Steam si non spécifiée.
     */
    fun getDistributionPlatform(): DistributionPlatform {
        return if (distributionPlatformId != null) {
            DistributionPlatform.fromId(distributionPlatformId) ?: DistributionPlatform.STEAM
        } else {
            DistributionPlatform.STEAM
        }
    }
}

/**
 * Représente la possession d'un jeu par un joueur.
 */
data class GameOwnership(
    val gameId: String,
    val gameName: String,
    val purchaseDate: String,
    val playtime: Int = 0, // en heures
    val lastPlayed: String? = null,
    val pricePaid: Double = 0.0
)

